src/lib.rs

```diff
@@ -27,10 +19,29 @@ macro_rules! try_opt_loc {
 
 pub struct Vfs<U = ()>(VfsInternal<RealFileLoader, U>);
 
-#[derive(Debug, Deserialize, Serialize)]
-pub struct Change {
-    pub span: Span,
-    pub text: String,
+pub type Changes = HashMap<PathBuf, FileChange>;
```

Comment:
I would prefer to keep changes as a list rather than a HashMap, since creating small hashmaps is not very efficient